govcd/vapp_network.go

```diff
@@ -223,3 +223,74 @@ func (vapp *VApp) RemoveAllNetworkFirewallRules(networkId string) error {
 	}
 	return nil
 }
+
+// UpdateNetworkStaticRouting updates vApp network static routes.
+// Returns pointer to types.VAppNetwork or error
+func (vapp *VApp) UpdateNetworkStaticRouting(networkId string, staticRoutes []*types.StaticRoute, enabled bool) (*types.VAppNetwork, error) {
+	task, err := vapp.UpdateNetworkStaticRoutingAsync(networkId, staticRoutes, enabled)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
+	}
+
+	return vapp.GetVappNetworkById(networkId, false)
+}
+
+// UpdateNetworkNatRulesAsync asynchronously updates vApp network static routes.
+// Returns task or error
+func (vapp *VApp) UpdateNetworkStaticRoutingAsync(networkId string, staticRoutes []*types.StaticRoute, enabled bool) (Task, error) {
+	util.Logger.Printf("[TRACE] UpdateNetworkStaticRoutingAsync with values: id: %s and staticRoutes: %#v, enable: %t", networkId, staticRoutes, enabled)
+
+	uuid := extractUuid(networkId)
+	networkToUpdate, err := vapp.GetVappNetworkById(uuid, true)
+	if err != nil {
+		return Task{}, err
+	}
+
+	if !isVappNetwork(networkToUpdate) {
+		return Task{}, fmt.Errorf("network static routing can be applied only for vapp network, not vapp org network")
+	}
+
+	if networkToUpdate.Configuration.Features == nil {
+		networkToUpdate.Configuration.Features = &types.NetworkFeatures{}
+	}
+	networkToUpdate.Xmlns = types.XMLNamespaceVCloud
+
+	networkToUpdate.Configuration.Features.StaticRoutingService = &types.StaticRoutingService{IsEnabled: enabled, StaticRoute: staticRoutes}
+
+	// here we use `PUT /network/{id}` which allow to change vApp network.
+	// But `GET /network/{id}` can return org VDC network or vApp network.
+	apiEndpoint := vapp.client.VCDHREF
+	apiEndpoint.Path += "/network/" + uuid
+
+	return vapp.client.ExecuteTaskRequest(apiEndpoint.String(), http.MethodPut,
+		types.MimeVappNetwork, "error updating vApp Network static routes: %s", networkToUpdate)
+}
+
+// Allows to identify if given network config is a vApp network and not a vApp Org network
+func isVappNetwork(networkConfig *types.VAppNetwork) bool {
```

Comment:
One consideration  - I think we have similar function in Terraform codebase. Wondering if they are 100% the same and if they are - maybe it is worth to make this function public and reuse in Terraform to avoid some flops of having slightly different checks in different places.

cc @dataclouder , @lvirbalas

Reply:
made public.